symengine/sets.h

```diff
@@ -34,7 +31,7 @@ class EmptySet : public Set {
     IMPLEMENT_TYPEID(EMPTYSET)
     //EmptySet(EmptySet const&) = delete;
     void operator=(EmptySet const&) = delete;
-    static RCP<const EmptySet> getInstance();
```

Comment:
I did this to generalise that `RCP <const Interval>` or `RCP <const EmptySet>` can only be generated by casting.
See that now, all the functions return `RCP <const Set>`